src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreBuildHandler.cs

```diff
@@ -117,6 +120,17 @@ public void Dispose()
 
         public void UpdateSolution_QueryDelayBuildAction(uint dwAction, out IVsTask pDelayTask)
         {
+            if (!_isMEFInitialized)
+            {
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
`RestoreAsync(...)` below accesses both imported fields (`Settings` and `SolutionRestoreWorker`).  Is `UpdateSolution_QueryDelayBuildAction(...)` guaranteed to execute before `RestoreAsync(...)` in all scenarios?  Otherwise those fields would be `null`.